### PR TITLE
LPS-154344 portal-search-web: Fix search insights' copy to clipboard on virtual instances

### PR DESCRIPTION
**Pull request by oliv-yu**
https://issues.liferay.com/browse/LPS-154344 - Search Insights widget "Copy to Clipboard" buttons not working on virtual instance

Solution was to import Clipboard https://clipboardjs.com/, which is done with the new `initializeClipboard.js` util file! 📄

### CODE REVIEW

File: modules/apps/portal-search/portal-search-web/src/main/resources/META-INF/resources/js/utils/initializeClipboard.js [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+		openToast({
+			message: Liferay.Language.get('an-unexpected-error-occurred'),
+			title: Liferay.Language.get('error'),
+			type: 'danger',
+		});
+	});
+
+	return () => {
+		clipboardJS.destroy();
+	};
```
[the remark this answers is not in the document]
Comment (oliv-yu) [reply modules/apps/portal-search/portal-search-web/src/main/resources/META-INF/resources/js/utils/initializeClipboard.js:39]: Hm maybe not actually. From some searching, I'm only seeing it in `useEffect` and I was being wishful. Sad.

I was trying to google a solution, what do you think of this stack overflow answer? https://stackoverflow.com/questions/34455807/using-clipboard-js-how-to-destroy-all-its-object-before-creating-new-one

Another way would be having it be a component? We could render a button with react and be able to use `useEffect`... let me know what you think?

[the remark this answers is not in the document]
Comment (thektan) [reply modules/apps/portal-search/portal-search-web/src/main/resources/META-INF/resources/js/utils/initializeClipboard.js:39]: I'm thinking we can add this listener:
```javascript
Liferay.once('beforeNavigate', () => {
   clipboardJS.destroy();
});
```

[the remark this answers is not in the document]
Comment (oliv-yu) [reply modules/apps/portal-search/portal-search-web/src/main/resources/META-INF/resources/js/utils/initializeClipboard.js:39]: ok! good idea 💯

**Review by thektan — APPROVED** [review]
Thanks for making that change!

**Comment by thektan** [comment]
ci:forward

**Pull request closed without merging**